pkgs/development/beam-modules/fetch-mix-deps.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenvNoCC, elixir, hex, rebar, rebar3, cacert }:
+
+{ name, version, sha256, src, mixEnv ? "prod", debug ? false, meta ? { } }:
+
+with stdenvNoCC.lib;
+
+stdenvNoCC.mkDerivation ({
+  name = "mix-deps-${name}-${version}";
+
+  phases = [ "configurePhase" "downloadPhase" ];
+
+  nativeBuildInputs = [ elixir hex cacert ];
+
+  inherit src;
+
+  MIX_ENV = mixEnv;
+  MIX_REBAR = "${rebar}/bin/rebar";
+  MIX_REBAR3 = "${rebar3}/bin/rebar3";
+  MIX_DEBUG = if debug then 1 else 0;
+
+  configurePhase = ''
+    mkdir -p $out/deps
+    mkdir -p $out/.hex
+    export HEX_HOME="$out/.hex";
+    export MIX_HOME="$TEMPDIR/.mix";
+    export MIX_DEPS_PATH="$out/deps";
+  '';
```

Comment:
There is a nice trick from `buildRustCrate`, in which which `fetchCargoTarball` copies the lock file inside `$out` so that the end derivation can [compare it](https://github.com/NixOS/nixpkgs/blob/d884b2d877cfda8247a6ee49f5c26ba3133c3889/pkgs/build-support/rust/default.nix#L164-L190) and verify if the `cargoSha256` is out of date. Maybe we could do that here?

Reply:
Very nice find!
Unless I'm wrong, there is no vendored step here that regenerates a mix.lock.
If I understand correctly, mix fetches dependencies from mix.lock. So there couldn't be a difference.
Happy to be proven wrong here.

Reply:
I think the idea behind that was not that there could be a vendored step that modifies the mix.lock, but that when upgrading a package one often forgets to modify the `mixSha256` (me definitely included). And since it's a fixed-output derivation and it's already in the nix store, nix will happily provide the wrong dependencies.

Reply:
I think I see what you mean. A `mixSha256` is passed, the content of the downloads should at least be checked against the sha256

Reply:
I just verified and the sha256 is being checked. As long as null is not supplied then the mixSha256 will be checked. Let me know if I misunderstood your comment

Reply:
Yes, I think we are talking about the same thing. If it is already checked by mix, then I guess there's no need to worry about it ^^

Reply:
Just to summarize and make sure we are on the same page.
There is only a single mix.lock and mix will get the related packages.
Nix will check that the provided sha256 matches against the dependency that are being fetched.
This can be tested with the instructions on this PR. Upon first fetch, nix will complain about the sha256